ui/js/search.js

```diff
@@ -0,0 +1,171 @@
+var SEARCH_COURSE_API = 'http://localhost:3000/search-course?searchQuery=';
+var COURSE_MATCHER = new RegExp("^[a-zA-Z]{3}[0-9]{1,3}$");
+var chosenCourses = [];
+
+$(document).ready(function () {
+    var input = '';
+
+    String.prototype.replaceAll = function (search, replacement) {
```

Comment:
I think this can be outside the `document.ready` function since it doesn't need the document to work.